addons/stock/models/stock_inventory.py

```diff
@@ -100,12 +106,19 @@ def _selection_filter(self):
     def onchange_filter(self):
         if self.filter not in ('product', 'product_owner'):
             self.product_id = False
+            self.exhausted =False
         if self.filter != 'lot':
             self.lot_id = False
         if self.filter not in ('owner', 'product_owner'):
             self.partner_id = False
         if self.filter != 'pack':
             self.package_id = False
+        if self.filter in ('pack','lot', 'product_owner', 'owner','partial'):
+            self.exhausted = False
+        if self.filter != 'category':
+            self.category_id = False
+        if self.filter == 'product':
+            self.exhausted = True
```

Comment:
If 'self.exhausted' is true only if "self.filter == 'product'", why not setting it to false by default and true only in this case ?